### ISSUE TITLE
CognitiveServices Not Targeted Preventing Build

### ISSUE THREAD

**Issue by alaskanrogue** [Azure-Samples/cognitive-services-speech-sdk#964]
Although the NuGet is installed and the .config file's packages section is modified, during the build, the library is not listed in the .targets file preventing its assembly from being loaded. The error description generated is:

``Error while loading assemblies: /Users/marcgeorge/Library/Caches/Xamarin/mtbs/builds/Sentry.iOS/f213a99dba3392a75f4a1f0a9727090befe053f16bf8a9eed002bd05e15a59a5/bin/iPhone/Debug/device-builds/iphone12.1-14.4/Microsoft.CognitiveServices.Speech.core.dll.	Sentry.iOS	C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\iOS	C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets	425	

Cognitive Services Speech SDK v1.15 is currently in use but the issue began with v1.14. 

The targeted platform is iPhone 11 with iOS14.4 using C# and Xamarin.

I have tried modifying the .csproj.nuget.g.targets file with the appropriate <Import> section, but it is removed during the build.

**Comment by jhakulin** [comment]
@alaskanrogue Hey Marc, not clearly seen from your message, what is the error you get while loading assemblies, is there more information to provide? Are you building your project on Visual Studio on Windows or Mac? Do you see a problem if you try with our Xamarin quickstart?
https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/quickstart/csharp/xamarin

**Comment by alaskanrogue** [comment]
@jhakulin  Jarno, the actual error is just "Error while loading assemblies:" with the subsequent path and reference to the Common.targets file and the error line in it. The error is during the build despite the "loading" statement. The DLL is not at the specified path. See attached. (Because of this, there is a "can't deploy" error.)
![Screenshot 2021-02-19 070207](https://user-images.githubusercontent.com/14001196/108502365-b00db200-7280-11eb-9efb-cdf6b47a9633.png)

**Comment by jhakulin** [comment]
@alaskanrogue We are not able to reproduce the issue on our Xamarin samples. Could you provide a small sample app which reproduces the issue? Currently we do not have enough information to proceed with the case.

**Comment by alaskanrogue** [comment]
@jhakulin Jarno, I created the sample app, but it is too large to attached. It was derived from the actual app so all configuration and referenced libraries are present. How should I provide it to you?

**Comment by alaskanrogue** [comment]
@jhakulin Jarno, further investigation after a build by going to the folder specified showed that the dll was not present.

**Comment by ICSentry** [comment]
@jhakulin Jarno, upgrading to Visual Studio 2019 version 16.9 Preview 5 had no positive impact.

**Comment by ICSentry** [comment]
@jhakulin @pankopon As I said Tuesday, I created a sample app, but it is too large to upload. How should I provide it to you for testing. It is 35MB.

**Comment by jhakulin** [comment]
@ICSentry Hi Marc, GitHub has apparently 25MB limit. You can either make sample which is smaller than 25MB or provide OneDrive link for your file share?

**Comment by alaskanrogue** [comment]
@jhakulin I divided the contents into two files with instructions to reassemble.

[Install.txt](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/6049448/Install.txt)
[Sentry.zip](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/6049451/Sentry.zip)
[Windows.Foundation.UniversalApiContract.zip](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/6049454/Windows.Foundation.UniversalApiContract.zip)

**Comment by alaskanrogue** [comment]
@jhakulin  There appears to be a dependency between Microsoft.CognitiveServices.Speech.core.dll and Microsoft.CognitiveServices.Speech.core.dll.a. I have seen several error messages referencing the later as not being present. Looking in the nugets repository you find this:
![Screenshot 2021-02-28 133924](https://user-images.githubusercontent.com/14001196/109429688-0b4f4b00-79cb-11eb-8cf6-44d7824b1e2e.png).

I found a native reference to it in the project. I don't know were it came from, but it wasn't in earlier working versions so I deleted it. I also deleted the nuget and reinstalled it. The .a file was there.  I subsequently did a build to encount the error again. Using SSH, I explored the Mac to the given error address and found the .dll present but not the .a file.

![Screenshot 2021-02-28 134005](https://user-images.githubusercontent.com/14001196/109429813-0e970680-79cc-11eb-9582-cfa72b188f54.png)

During the build I did diagnostic logging and at the end of the log I found this:

`2>  /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/bin/mtouch @/Users/marcgeorge/Library/Caches/Xamarin/mtbs/builds/Sentry.iOS/f213a99dba3392a75f4a1f0a9727090befe053f16bf8a9eed002bd05e15a59a5/obj/iPhone/Debug/response-file.rsp "--gcc_flags=-LC:/Users/Rogue.THOUGHTS/.nuget/packages/microsoft.cognitiveservices.speech/1.15.0/build/Xamarin.iOS -lMicrosoft.CognitiveServices.Speech.core -force_load C:/Users/Rogue.THOUGHTS/.nuget/packages/microsoft.cognitiveservices.speech/1.15.0/build/Xamarin.iOS//libMicrosoft.CognitiveServices.Speech.core.a -LC:/Users/Rogue.THOUGHTS/.nuget/packages/microsoft.cognitiveservices.speech/1.15.0/build/Xamarin.iOS -lMicrosoft.CognitiveServices.Speech.core -force_load C:/Users/Rogue.THOUGHTS/.nuget/packages/microsoft.cognitiveservices.speech/1.15.0/build/Xamarin.iOS//libMicrosoft.CognitiveServices.Speech.core.a"  (TaskId:279)
2>  Xamarin.iOS 14.14.2.5 (d16-9: 3836759d4 using framework: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.4.sdk (TaskId:279)
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(425,3): error MT0009: Error while loading assemblies: /Users/marcgeorge/Library/Caches/Xamarin/mtbs/builds/Sentry.iOS/f213a99dba3392a75f4a1f0a9727090befe053f16bf8a9eed002bd05e15a59a5/bin/iPhone/Debug/Microsoft.CognitiveServices.Speech.core.dll.
2>  		 (TaskId:279)
2>  The command exited with code 1. (TaskId:279)
2>  MTouch: 2021-02-28T13:13:05.5021770-05:00 - Finished (TaskId:279)
2>Done executing task "MTouch" -- FAILED. (TaskId:279)
2>Done building target "_CompileToNative" in project "Sentry.iOS.csproj" -- FAILED.: (TargetId:326)
`
Basically, why isn't the .a file not being copied to the Debug folder?

**Comment by alaskanrogue** [comment]
@jhakulin Do you know what the role of the ".a" file as a native reference? It is pretty sizable. Is it a library for the ".dll"?

**Comment by alaskanrogue** [comment]
The problem exists in VS v 16.10.0 Preview 1.0.

**Comment by alaskanrogue** [comment]
@jhakulin Any progress?

**Comment by jhakulin** [comment]
@alaskanrogue libMicrosoft.CognitiveServices.Speech.core.a is used on iOS build. There is no dependency to Microsoft.CognitiveServices.Speech.core.dll, which is used in Windows UWP builds. The problem seems to be in your \Sentry.iOS\Sentry.iOS.csproj file. After modifying it to the attached version, Sentry builds ok, with your original project file, I can reproduce the same problem you are seeing. Could you let me know which Visual Studio project template you were using in your project? 
[Sentry.iOS.csproj.zip](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/6268662/Sentry.iOS.csproj.zip)

**Comment by jhakulin** [comment]
closing as solution provided to customer. Let us know if there is any further issue.

**Comment by alaskanrogue** [comment]
@jhakulin I have been traveling for the past several weeks. I had the opportunity to remotely re-integrate your .csproj file today, adding the configuration that I had deleted for your testing project. I was able to remotely, successfully build the solution, However I am unable to test until Monday when I am back at my office,

Sorry, but I am unable to advise you to the project template used. It wasn't recent. It may date back to the original MS version. I migrated from the original pre-MS versions of Xamarin.
